File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/BatteryEvent.java
package com.mapbox.services.android.navigation.v5.internal.navigation;

import android.annotation.SuppressLint;
import android.os.Parcelable;

@SuppressLint("ParcelCreator")
@SuppressWarnings("ParcelableCreator")
class BatteryEvent extends NavigationPerformanceEvent implements Parcelable {
  private static final String BATTERY_PERCENTAGE_KEY = "battery_percentage";
  private static final String IS_PLUGGED_IN_KEY = "is_plugged_in";
  private static final String BATTERY_EVENT_NAME = "battery_event";

  BatteryEvent(String sessionId, float batteryPercentage, boolean isPluggedIn,
               NavigationPerformanceMetadata metadata) {
    super(sessionId, BATTERY_EVENT_NAME, metadata);

    addCounter(new FloatCounter(BATTERY_PERCENTAGE_KEY, batteryPercentage));
    addAttribute(new Attribute(IS_PLUGGED_IN_KEY, String.valueOf(isPluggedIn)));
  }
}

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/internal/navigation/BatteryEvent.kt
package com.mapbox.services.android.navigation.v5.internal.navigation

import android.annotation.SuppressLint
import android.os.Parcelable

@SuppressLint("ParcelCreator")
internal data class BatteryEvent(
    private val sessionId: String,
    private val batteryPercentage: Float,
    private val isPluggedIn: Boolean,
    override var metadata: NavigationPerformanceMetadata?
) : NavigationPerformanceEvent(sessionId, BATTERY_EVENT_NAME, metadata), Parcelable {

    companion object {
        private const val BATTERY_PERCENTAGE_KEY = "battery_percentage"
        private const val IS_PLUGGED_IN_KEY = "is_plugged_in"
        private const val BATTERY_EVENT_NAME = "battery_event"
    }

    init {
        addCounter(
            FloatCounter(
                BATTERY_PERCENTAGE_KEY,
                batteryPercentage
            )
        )
        addAttribute(
            Attribute(
                IS_PLUGGED_IN_KEY,
                isPluggedIn.toString()
            )
        )
    }
}
